tests/lib/rules/no-implicit-globals.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * @fileoverview Tests for no-implicit-globals rule.
+ * @author Joshua Peek
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+var rule = require("../../../lib/rules/no-implicit-globals"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+var ruleTester = new RuleTester();
+ruleTester.run("no-implicit-globals", rule, {
+    valid: [
+        {
+            code: "window.foo = 1;"
+        },
+        {
+            code: "window.foo = function() {};"
+        },
+        {
+            code: "window.foo = function foo() {};"
+        },
+        {
+            code: "self.foo = 1;"
+        },
+        {
+            code: "self.foo = function() {};"
+        },
+        {
+            code: "this.foo = 1;"
+        },
+        {
+            code: "this.foo = function() {};"
+        },
+        {
+            code: "Utils.foo = 1;"
+        },
+        {
+            code: "Utils.foo = function() {};"
+        },
+        {
+            code: "(function() { var foo = 1; })();"
+        },
+        {
+            code: "(function() { let foo = 1; })();",
+            ecmaFeatures: { blockBindings: true }
+        },
+        {
+            code: "(function() { const foo = 1; })();",
+            ecmaFeatures: { blockBindings: true }
+        },
+        {
+            code: "(function() { function foo() {} })();"
+        },
+        {
+            code: "foo = 1;",
+            ecmaFeatures: { modules: true }
```

Comment:
That's because it's not an implicit global, it's a runtime error. Should still be reported by eslint, but not sure if this rule is the most appropriate place.